### PR TITLE
Call sparql for the first time

### PR DESCRIPTION
**Pull request by bastbijl**
Perform a narql query

I certify that:
- [x] The automated build passes on the branch this pull request is from
- [x] My change is documented in the Changelist.md
- [ ] Any functionality additions/changes are documented in the README.md
- [ ] Any functionality additions/changes are _also_ documented [here](https://github.com/weaverplatform/weaver-docs/blob/master/pages/developers/reference/weaver-sdk-js.md "Weaver-docs")
- [x] Any new dockerimages this build depends on are properly tagged on github
- [x] I shall not merge until the travis build is green and the PR is accepted by
  another developer
- [x] I shall promptly fix/reply to comments, as I know PRs are not :fire: and
  forget
- [x] I shall keep my branch updated with any intermediate changes to the
  target branch
- [x] This is a quality PR
- [x] I got to the end of this list

### CODE REVIEW

File: test/WeaverNarql.test.coffee [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      wipeCurrentProject().then( ->
+        Promise.all([a.save(), b.save(), c.save(), d.save()])
+      ).then( ->
+        a.destroy()
+      ).then( ->
+        c.relation('to').remove(x)
+      )
+
+    it 'should for partly deleted nodes do a simple narql query', ->
+      new Weaver.Narql('_ ?x to ?y .')
```
Comment (gvdent) [inline test/WeaverNarql.test.coffee]: Why would the `.` be required here?

File: test/WeaverNarql.test.coffee [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        .restrict(res.x)
+        .find()
+      ).then((res) ->
+        map = {}
+        map[node.id()] = node.relation('to').first().id() for node in res
+        assert.deepEqual({ b: 'x', d: 'x' }, map)
+      )
+
+  describe 'a cycle of a certain size', ->
+    certainSize = 7
```
Comment (gvdent) [inline test/WeaverNarql.test.coffee]: This tests the edge case that the cycle is 'long', please also test certainSize 2 to ensure short cycles work.

File: test/WeaverNarql.test.coffee [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Weaver = require('../src/Weaver')
+
+checkNodeInResult = (nodes, nodeId) ->
+  ids = (i.id() for i in nodes)
+  expect(ids).to.contain(nodeId)
+
+describe 'WeaverQuery Narql Test', ->
+
+  describe 'a simple test set', ->
+    x = new Weaver.Node("x")
```
Comment (gvdent) [inline test/WeaverNarql.test.coffee]: Please use single quoted strings for non interpolated strings.

File: test/WeaverNarql.test.coffee [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    c.relation('to').add(x)
+    d.relation('to').add(x)
+
+    before ->
+      wipeCurrentProject().then( ->
+        Promise.all([a.save(), b.save(), c.save(), d.save()])
+      ).then( ->
+        a.destroy()
+      ).then( ->
+        c.relation('to').remove(x)
```
Comment (gvdent) [inline test/WeaverNarql.test.coffee]: Please also add the following:
- A relation 'to' that first targetted 'x' replaced with one that points to 'y'
- A relation 'to' on the relation 'e' -to-> 'x', with target 'x'
- A relation 'f' -notto-> 'x'
- A relation 'x' -to-> 'z'

File: test/WeaverNarql.test.coffee [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        for node in res
+          number = parseInt(node.id())
+          next = parseInt(node.relation('next').first().id())
+          previous = parseInt(node.relation('previous').first().id())
+          assert.equal((number+1)%cycle.length, next)
+          assert.equal((cycle.length+number-1)%cycle.length, previous)
+      )
+
+    it 'should find one node from a cycle', ->
+      new Weaver.Narql('_ ?x next \'4\' .')
```
Comment (gvdent) [inline test/WeaverNarql.test.coffee]: Ugh, avoid escaping at all costs, `"_ ?x next '4' ."` is neater.

**Comment by gvdent** [comment]
This should not be merged back before the narql functionality has been merged back to develop on both weaver-database-postgresql and weaver-server
